File: python-algo/leetcode/problem_476.py
"""476. Number Complement
https://leetcode.com/problems/number-complement/
"""


class Solution:
    def findComplement(self, num: int) -> int:
        ans = 0
        i = 0
        while num:
            bit = num & 0b1
            if bit == 0:
                ans += 1 << i
            num = num >> 1
            i += 1
        return ans

File: java-algo/src/main/java/com/leetcode/Problem476.java
package com.leetcode;

/**
 * 476. Number Complement
 * https://leetcode.com/problems/number-complement/
 */
public class Problem476 {
    public int findComplement(int num) {
        int ans = 0, i = 0;
        while (num != 0) {
            int bit = 1 - (num & 1);
            ans += (bit << i);
            num >>= 1;
            i++;
        }
        return ans;
    }
}
